src/gui/main.cpp

```diff
@@ -172,7 +183,32 @@ void MyFrame::onClickDisconnect(wxCommandEvent &event)
 }
 */
 
+void MyFrame::onClose(wxCloseEvent &event)
+{
+	if (systray_state) {
+		this->Hide();
+		systray_state = false;
+	} else {
+		this->Show();
+		systray_state = true;
+	}
+
+	event.Veto();
+}
+
 MyFrame *frame;
+wxMenu *MyTaskBarIcon::CreatePopupMenu()
```

Comment:
hmm sorry for the many questions but when is this exactly called? Didn't see this explicitly Connected above.

Reply:
Right, there is a bit of magic here. So there is this new class I created
`class MyTaskBarIcon: public wxTaskBarIcon`
So this class already call this internal function when we click on the icon, but the function in the class is declared "virtual" so we can override it and change the behavior:
https://docs.wxwidgets.org/3.0/classwx_task_bar_icon.html#a739c81dc5cadf9e550d38001cee419f6

So we don't really need to Connect() the action to a callback, since it was implicit in that case.

Reply:
ahh gotcha makes sense!